app/assets/javascripts/discourse/app/components/composer/textarea-editor.gjs

```diff
@@ -0,0 +1,85 @@
+import Component from "@glimmer/component";
+import { getOwner } from "@ember/owner";
+import { modifier } from "ember-modifier";
+import DTextarea from "discourse/components/d-textarea";
+import TextareaTextManipulation from "discourse/lib/textarea-text-manipulation";
+import { bind } from "discourse-common/utils/decorators";
+
+export default class TextareaEditor extends Component {
+  textarea;
+
+  registerTextarea = modifier((textarea) => {
+    this.textarea = textarea;
+
+    this.textManipulation = new TextareaTextManipulation(getOwner(this), {
+      markdownOptions: this.args.markdownOptions,
+      textarea,
+    });
+
+    const destructor = this.args.onSetup(this.textManipulation);
```

Comment:
Why do we need this const here? and not directly call onSetup in the return block

Reply:
`onSetup` needs to happen outside the return block, because it's being used to set the `textManipulation` in d-editor

Reply:
Ah yes right, sorry it was stupid